### PR TITLE
Add config for max steps per iteration

### PR DESCRIPTION
**Pull request by jamesbornholt**
This change adds a configuration for an upper bound on the number of schedule steps
a single test iteration can take. When the bound is hit, the config controls whether
the test is considered failed or whether to continue to the next iteration (i.e., to
abort the current one and move on). By default, I'm setting the behavior to fail the
test if 1,000,000 steps are taken, so we can get earlier warnings about fairness issues
in Shuttle tests.

The DFS scheduler already had an option like this, which I've now removed and replaced
with the general config mechanism. Schedulers can still return `None` as the next task
to end an iteration early, though that functionality isn't used any more.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/scheduler/dfs.rs [hunk 1]
```diff
@@ -21,18 +20,15 @@ pub struct DfsScheduler {
...
 
 impl DfsScheduler {
-    /// Construct a new DFSScheduler with an optional bound on how many iterations to run
-    /// and an optional bound on the maximum depth to explore. A DFSScheduler can optionally
-    /// allow random data to be generated by the test (using `shuttle::rand`). Enabling random
-    /// data makes the DFS search incomplete, as it will not explore all possible values for
-    /// the random choices. To ensure determinism, each execution will use the same sequence of
-    /// random choices.
-    pub fn new(max_iterations: Option<usize>, max_depth: Option<usize>, allow_random_data: bool) -> Self {
+    /// Construct a new DFSScheduler. A DFSScheduler can optionally allow random data to be
```
Comment (jorajeev) [inline src/scheduler/dfs.rs:23]: Could you add back the comment about the `max_iterations` argument?

File: src/scheduler/mod.rs [hunk 2]
```diff
@@ -12,6 +11,8 @@ mod round_robin;
 pub(crate) mod metrics;
 pub(crate) mod serialization;
 
+pub use crate::runtime::task::TaskId;
```
Comment (jorajeev) [inline src/scheduler/mod.rs:14]: Why make this pub?

File: src/scheduler/dfs.rs (hunk 1 above)
Comment (jamesbornholt) [reply src/scheduler/dfs.rs]: Oops, done

File: src/scheduler/mod.rs (hunk 2 above)
Comment (jamesbornholt) [reply src/scheduler/mod.rs:14]: It's to be able to implement the `Scheduler` trait, which is also pub, in client code (in this case, in a unit test). I'm actually surprised the compiler didn't complain about it before.

**Review by jorajeev — APPROVED** [review]

**Pull request merged**